L26-Ability-to-restrict-max-threads-in-C++SyncServer.md

```diff
@@ -0,0 +1,93 @@
+* Author(s): Sree Kuchibhotla (sreecha)
+* Approver: vjpai
+* Status: In Review
+* Implemented in: C++
+* Last updated: June 6, 2018
+* Discussion at: 
+
+## Abstract
+C++ `ThreadManager` is a specialized thread pool used to implement the C++ Synchronous server. 
+
+Currently, `ThreadManager` can be configured to have minimum and maximum number of polling threads. While this helps to always have threads available to poll for work, it does nothing to throttle polling if the `ThreadManager` is overloaded with work.
+
+The proposal here is to add a notion of 'thread quota' to the `resource_quota` object. Currently, a resource quota object can be attached to the server (via `ServerBuilder::SetResourceQuota`). The idea here is to set a maximum number of threads on that resource quota object.
+
+Each of the `ThreadManager` objects in a given server check with the Server's resource quota object when creating new threads and will not create new threads if no quota is available.
+
+## Details
+More concretely, the following API changes are being proposed
```

Comment:
Can you take the code details out of the document and put them in a PR linked from the document instead? I like to add that in the "Implemented In" line in the header

Reply:
o you mean the private API details ?  - the other code snippets are public API changes and I thought should be in this document..

There is no PR for this yet (I mean, I have an in-progress change in my local branch but I did not implement the bulk of the changes yet. I want to get some agreement on the API before proceeding.